src/runtime/swg-client-event-manager-test.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2018 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnalyticsEvent,EventOriginator} from '../proto/api_messages';
+import * as EventManagerApi from '../api/swg-client-event-manager-api';
+import {SwgClientEventManager} from './swg-client-event-manager';
+
+const DEFAULT_TYPE = AnalyticsEvent.IMPRESSION_AD;
+const DEFAULT_ORIGIN = EventOriginator.SWG_CLIENT;
+const OTHER_TYPE = AnalyticsEvent.ACTION_PAYMENT_COMPLETE;
+const OTHER_ORIGIN = EventOriginator.AMP_CLIENT;
+const BAD_VALUE = 'I should throw an error';
+
+/** @type {!EventManagerApi.SwgClientEvent} */
+const DEFAULT_EVENT = {
+  eventType: DEFAULT_TYPE,
+  eventOriginator: DEFAULT_ORIGIN,
+  isFromUserAction: null,
+  additionalParameters: {},
+};
+
+describes.sandboxed('EventManager', {}, () => {
+  it('should throw an error for invalid events', () => {
+    /** @type {!EventManagerApi.SwgClientEvent} */
+    let event = {
+      eventType: DEFAULT_TYPE,
+      eventOriginator: DEFAULT_ORIGIN,
+      isFromUserAction: null,
+      additionalParameters: {},
+    };
+    const eventMan = new SwgClientEventManager();
+
+    let errorCount = 0;
+    const tryIt = () => {
+      try {
+        eventMan.logEvent(event);
+      } catch (e) {
+        errorCount++;
```

Comment:
do you not intend to validate the error message as well?

Reply:
Ok, done